gbpservice/neutron/nsf/core/main.py

```diff
@@ -0,0 +1,705 @@
+                    % (el.identify(), event.identify(), el.identify())))
+                remevs.append(el)
+        self._cache.rem_multi(remevs)
+
+    def _get(self):
+        try:
+            return self._pollq.get(timeout=0.1)
+        except Empty:
+            return None
+
+    def fill(self):
+        LOG.debug(_("Fill events from multi processing Q to internal cache"))
+        # Get some events from queue into cache
+        for i in range(0, 10):
+            ev = self._get()
+            if ev:
+                LOG.debug(_(
+                    "Got new event %s from multi processing Q"
+                    % (ev.identify())))
+                self._cache.add(ev)
+
+    def peek(self, idx, count):
+        LOG.debug(_("Peek poll events from index:%d count:%d" % (idx, count)))
+        cache = self._cache.copy()
+        qlen = len(cache)
+        LOG.debug(_("Number of elements in poll q - %d" % (qlen)))
+        pull = qlen if (idx + count) > qlen else count
+        return cache[idx:(idx + pull)], pull
+
+    def event_done(self, ev):
+        LOG.info(_("Poll event %s to be marked done !" % (ev.identify())))
+        self.rem(ev)
+
+    def _cancelled(self, ev):
+        LOG.info(_("Poll event %s cancelled" % (ev.identify())))
+        ev.poll_event = 'POLL_EVENT_CANCELLED'
+        self.event_done(ev)
+        self._sc.rpc_event(ev)
+
+    def _sched(self, ev):
+        LOG.debug(_("Schedule event %s" % (ev.identify())))
+        eh = self._eh.get(ev)
+        if isinstance(eh, core_periodic_task.PeriodicTasks):
+            if eh.check_timedout(ev):
+                LOG.info(_(
+                    "Event %s timed out -"
+                    "scheduling it to a worker" % (ev.identify())))
+                self._sc.rpc_event(ev)
+                return ev
+        else:
+            LOG.info(_(
+                "Event %s timed out -"
+                "scheduling it to a worker" % (ev.identify())))
+            self._sc.rpc_event(ev)
+            return ev
+        return None
+
+    def event(self, ev):
+        ev1 = copy.deepcopy(ev)
+        ev1.serialize = False
+        ev1.poll_event = \
+            'POLL_EVENT_CANCELLED' if ev1.max_times == 0 else 'POLL_EVENT'
+        if ev1.poll_event == 'POLL_EVENT_CANCELLED':
+            self._cancelled(ev1)
+        else:
+            if self._sched(ev1):
+                ev.max_times -= 1
+                ev.last_run = ev1.last_run
+
+    def process(self, ev):
+        LOG.debug(_("Processing poll event %s" % (ev.identify())))
+        self.event_done(ev) if ev.id == 'POLL_EVENT_DONE' else self.event(ev)
+
+    def poll(self):
+        # Fill the cache first
+        self.fill()
+        # Peek the events from cache
+        evs, count = self.peek(0, self._batch)
+        for ev in evs:
+            self.process(ev)
+        self._procidx = (self._procidx + count) % (self._batch)
+
+
+class EventHandler(object):
+
+    def __init__(self, sc, qu, eh):
```

Comment:
Using 'eh' and '_eh' below makes this hard to read. This class is EventHandler (eh) and you are actually passing in a event_desc_table. As noted above EventHandler should be renamed since it's processing the event_queue. eh, _eh are actually event_desc_tables?